qcfractal/storage_sockets/sqlalchemy_socket.py

```diff
@@ -1153,15 +1156,24 @@ def add_results(self, record_list: List[ResultRecord]):
 
                 if get_count_fast(doc) == 0:
                     doc = ResultORM(**result.dict(exclude={"id"}))
-                    session.add(doc)
-                    session.commit()  # TODO: faster if done in bulk
-                    result_ids.append(str(doc.id))
+                    # results_list.append(doc)
+                    results_list.append(doc)
+                    results_idx.append(i)
+                    # session.commit()  # TODO: faster if done in bulk
+                    # result_ids.append("placeholder")
                     meta["n_inserted"] += 1
                 else:
                     id = str(doc.first().id)
                     meta["duplicates"].append(id)  # TODO
                     # If new or duplicate, add the id to the return list
-                    result_ids.append(id)
+                    result_ids[i] = id
+
+            session.add_all(results_list)
+            session.commit()
+            for i, res in enumerate(results_list):
```

Comment:
The changes to `add_results` are not faster since we will need to query every loop which is the expensive part.

Reply:
```python
cond1 = [Mymodel.basis=='A', Mymodel.driver=='B']
cond1 = [Mymodel.basis=='C', Mymodel.driver=='D']
Mymodel.query.filter(or_(and_(*cond1), and_(*cond2)))
```

This may work to query all multi-indices at the same time. Thanks @doaa-altarawy!

Reply:
Thanks for your input @doaa-altarawy ! I tried using `tuple_` and `in_` first which seemed more convenient, but had to fall back to this solution since `in_` does not translate `None` to `IS NULL`. The change is brought in the new code, along with new comparison.

Reply:
Yea the choice for keywords to be a true `NULL` may need to be revisited in the future. We may want a "blank" Keywords entry that effectively returns `NULL`